### PR TITLE
Enos connection issue

### PR DESCRIPTION
**Pull request by amuraleedhar**
SUMMARY

Resolving connection issues of ansible 

ISSUE TYPE

Bugfix Pull Request

COMPONENT NAME

lib/ansible/modules/network/enos/enos_facts.py
lib/ansible/module_utils/enos.py
lib/ansible/plugins/action/enos.py
lib/ansible/plugins/terminal/enos.py

ANSIBLE VERSION

ansible 2.5.0
config file = /etc/ansible/ansible.cfg
configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
ansible python module location = /usr/local/lib/python2.7/dist-packages/ansible-2.5.0-py2.7.egg/ansible
executable location = /usr/local/bin/ansible
python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]

ADDITIONAL INFORMATION

This is an effort to create modules for enos based switches from Lenovo.

### CODE REVIEW

File: lib/ansible/module_utils/enos.py [hunk 1]
```diff
@@ -144,8 +145,12 @@ def run_commands(module, commands, check_rc=True):
 
 
 def load_config(module, config):
-    conn = get_connection(module)
-    conn.edit_config(config)
+    try:
+        conn.edit_config(config)
```
Comment (kedarkekan) [inline lib/ansible/module_utils/enos.py]: you can delete line 149

File: lib/ansible/plugins/terminal/enos.py [hunk 2]
```diff
@@ -76,7 +76,7 @@ def on_authorize(self, passwd=None):
             # Note: python-3.5 cannot combine u"" and r"" together.  Thus make
             # an r string and use to_text to ensure it's text
             # on both py2 and py3.
-            cmd[u'prompt'] = to_text(r"[\r\n]?password: $",
+            cmd[u'prompts] = to_text(r"[\r\n]?password: $",
```
Comment (kedarkekan) [inline lib/ansible/plugins/terminal/enos.py]: did you mean `cmd[u'prompts']`? missing a closing quote

Comment (gundalow) [inline lib/ansible/plugins/terminal/enos.py]: Is this right?
`grep -r "u'prompts'" lib/ansible` -> zero results
`grep -r "u'prompt'" lib/ansible` -> many results

Comment (kedarkekan) [reply lib/ansible/plugins/terminal/enos.py]: +1, @amuraleedhar pls check your command spec for whether `prompt` or `prompts` is the correct key.

**Comment by gundalow** [comment]
Regarding `prompt` vs `prompts` https://github.com/ansible/ansible/pull/32899 fixes a typo in network_cli

**Comment by amuraleedhar** [comment]
Please merge this. Thanks
-Anil

**Pull request merged**